### PR TITLE
Use django-storages' manifest files class instead of the overriden class

### LINKED ISSUE #8773: Use `django-storages`'s manifest files class
`django-storages` already implemented the manifest files class (see https://github.com/jschneier/django-storages/issues/733#issuecomment-989005340) so we can use it directly and remove our override for this in:

https://github.com/readthedocs/readthedocs.org/blob/933c316f5228423f9f871b3da858a01491f79925/readthedocs/storage/s3_storage.py#L55-L61

### PR DESCRIPTION
**Pull request by dogukanteber**
Hopefully resolves #8773

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Thanks for your contribution @dogukanteber ! We will have a look at it shortly

File: readthedocs/settings/docker_compose.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def show_debug_toolbar(request):
     # Storage backend for build languages
     RTD_BUILD_TOOLS_STORAGE = 'readthedocs.storage.s3_storage.S3BuildToolsStorage'
     # Storage for static files (those collected with `collectstatic`)
-    STATICFILES_STORAGE = 'readthedocs.storage.s3_storage.S3StaticStorage'
+    STATICFILES_STORAGE = 'storages.backends.s3boto3.S3ManifestStaticStorage'
```
Comment (humitos) [inline readthedocs/settings/docker_compose.py:158]: We shouldn't use this backend directly. Instead, we should keep using the one we were using but inheriting from this one that support manifest instead.

In our own backed we are using other stuff as well.

File: readthedocs/storage/s3_storage.py [hunk 2]
```diff
@@ -19,23 +19,6 @@
 from .mixins import OverrideHostnameMixin, S3PrivateBucketMixin
 
 
-class S3BuildMediaStorage(BuildMediaStorageMixin, OverrideHostnameMixin, S3Boto3Storage):
```
Comment (humitos) [inline readthedocs/storage/s3_storage.py:22]: It seems that you are deleting the wrong backend here. This one is for `BuildMedia`, and we need to modify the one for static files.

**Review by humitos — CHANGES_REQUESTED** [review]
Thanks for your PR. I added some comments that need to be addressed before moving forward.

**Comment by dogukanteber** [comment]
Thank you for your comments @humitos. I will make the requested changes.

File: readthedocs/storage/s3_storage.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
 from django.conf import settings
 from django.contrib.staticfiles.storage import ManifestFilesMixin
```
Comment (humitos) [inline readthedocs/storage/s3_storage.py:13]: I think this is not used anymore, is that correct? In that case, we can delete this import from here.

**Review by humitos — APPROVED** [review]
This looks good to me! Thanks @dogukanteber! We should be able to merge this PR and tested locally by running `collectstatic` 💪🏼

File: readthedocs/storage/s3_storage.py (hunk 3 above)
Comment (dogukanteber) [reply readthedocs/storage/s3_storage.py]: Yeah, I missed that. I deleted it.

**Comment by dogukanteber** [comment]
@humitos I guess something went wrong during CircleCI tests. I have been waiting the tests results for half an hour but nothing came up. So I took a look at the details and saw this error message:
```GLOB sdist-make: /home/circleci/project/setup.py
sphinx-18 create: /home/circleci/project/.tox/sphinx-18
sphinx-18 installdeps: Sphinx~=1.8.0
ERROR: Got SIGTERM, handling it as a KeyboardInterrupt
ERROR: got KeyboardInterrupt signal


Too long with no output (exceeded 10m0s): context deadline exceeded
```

Did I cause this error? If not, can we rerun the tests?

**Comment by stsewd** [comment]
> @dogukanteber I'm not sure what's happening with CircleCI. I'm having the same problem with my PRs and I wasn't able to fix it yet. @stsewd do you know what it could be?

nope

**Comment by dogukanteber** [comment]
Hello @humitos. Any improvements about the PR?

**Comment by humitos** [comment]
I think the problem with CircleCI was an issue with pip. I re-triggered the tests to be run again. Let's see how it goes. If that does not work, you could rebase this PR against master or merge master into it.

**Comment by dogukanteber** [comment]
I have merged master into this branch @humitos. It seems the previous error is disappeared. However, there is a Pylint error that says ```pylint: too-many-ancestors / Too many ancestors (9/7)```. I did not want to sweep under the rug just to get rid of the error. I'd like to get your opinion on this. How should we proceed?

**Comment by humitos** [comment]
😓 --you can disable that error by adding a comment right next the class definition:

```
class S3...(.., ..):  # pylint: disable=too-many-ancestors
```

**Comment by dogukanteber** [comment]
I saw that coming. Now it says ```E501 / line too long (124 > 100 characters)```. I do not how to get around this error without removing the pylint error @humitos.

**Comment by humitos** [comment]
You can use

```
class MyClass(
   AnotherClass,
   AnotherClassExtra,
   ...
):
```

to avoid the line to be too long.

**Comment by dogukanteber** [comment]
I suppose, I finally did it @humitos. I hope we can merge this in.

**Comment by humitos** [comment]
Yeaahhh! Awesome! Thank you so much! 💪

**Pull request merged**